src/auth/ha-auth-flow.ts

```diff
@@ -288,6 +302,57 @@ class HaAuthFlow extends litLocalizeLiteMixin(LitElement) {
     }
   }
 
+  private _externalStep(step: DataEntryFlowStepExternal) {
+    const external = window.open(step.url);
+    if (external) {
+      window.addEventListener(
+        "message",
+        async (message: MessageEvent) => {
+          if (
+            message.data.type === "externalCallback" &&
+            message.source === external
+          ) {
+            const postData = { ...this._stepData, client_id: this.clientId };
+
+            try {
+              const response = await fetch(`/auth/login_flow/${step.flow_id}`, {
+                method: "POST",
+                credentials: "same-origin",
+                body: JSON.stringify(postData),
+              });
+
+              const newStep = await response.json();
+
+              if (response.status === 403) {
+                this._state = "error";
+                this._errorMessage = newStep.message;
+                return;
+              }
+
+              if (newStep.type === "create_entry") {
+                this._redirect(newStep.result);
+                return;
+              }
+              this._step = newStep;
+              this._state = "step";
+            } catch (err) {
+              // eslint-disable-next-line no-console
+              console.error("Error handling external flow", err);
+              this._state = "error";
+              this._errorMessage = this._unknownError();
+            }
+          }
+        },
+        {
+          once: true,
+        }
+      );
+    } else {
+      this._state = "error";
+      this._errorMessage = "Login window was blocked, please allow popups.";
```

Comment:
Give the user the option to open it manually, browsers often do allow opening a window with user interaction